packages/build-tools/src/ios/configure.ts

```diff
@@ -34,4 +62,68 @@ async function configureXcodeProject<TJob extends Ios.Job>(
   }
 }
 
+async function updateVersionsAsync(
+  ctx: BuildContext<Ios.Job>,
+  {
+    targetNames,
+    buildConfiguration,
+  }: {
+    targetNames: string[];
+    buildConfiguration: string;
+  }
+): Promise<void> {
+  const project = IOSConfig.XcodeUtils.getPbxproj(ctx.reactNativeProjectDirectory);
+  const iosDir = path.join(ctx.reactNativeProjectDirectory, 'ios');
+
+  const infoPlistPaths: string[] = [];
+  for (const targetName of targetNames) {
+    const xcBuildConfiguration = IOSConfig.Target.getXCBuildConfigurationFromPbxproj(project, {
+      targetName,
+      buildConfiguration,
+    });
+    const infoPlist = xcBuildConfiguration.buildSettings.INFOPLIST_FILE;
+    if (infoPlist) {
+      const evaluatedInfoPlistPath = trimQuotes(
+        evaluateTemplateString(infoPlist, {
+          SRCROOT: iosDir,
+        })
+      );
+      const absolutePath = path.isAbsolute(evaluatedInfoPlistPath)
+        ? evaluatedInfoPlistPath
+        : path.join(iosDir, evaluatedInfoPlistPath);
+      infoPlistPaths.push(path.normalize(absolutePath));
+    }
+  }
+  const uniqueInfoPlistPaths = uniq(infoPlistPaths);
+  for (const infoPlistPath of uniqueInfoPlistPaths) {
+    ctx.logger.info(`Updating versions in ${infoPlistPath}`);
+    const infoPlistRaw = await fs.readFile(infoPlistPath, 'utf-8');
+    const infoPlist = plist.parse(infoPlistRaw) as IOSConfig.InfoPlist;
+    if (ctx.job.version?.buildNumber) {
+      infoPlist.CFBundleShortVersionString = ctx.job.version?.buildNumber;
+    }
+    if (ctx.job.version?.version) {
+      infoPlist.CFBundleVersion = ctx.job.version?.version;
+    }
+    await fs.writeFile(infoPlistPath, plist.build(infoPlist));
+  }
```

Comment:
Use `Promise.all`.

Reply:
I would prefer not to do that, in most cases it will be just one file, so there is not much performance input, and if something goes wrong we will know where it failed

Reply:
You haven't convinced me. With `Promise.all` you also know what was the problem :) You just need to throw an error with a meaningful error message.